tests/unit/Core/Content/Media/Infrastructure/Path/MediaUrlGeneratorTest.php

```diff
@@ -27,6 +28,25 @@ public function testGenerate(UrlParams $params, ?string $expected): void
             new Filesystem(new InMemoryFilesystemAdapter(), ['public_url' => 'http://localhost:8000']),
         );
 
+        if (Feature::isActive('v6.8.0.0') && $expected !== null) {
+            $urlParts = parse_url($expected);
+
+            if ($urlParts === false) {
+                throw new \InvalidArgumentException('Invalid URL given for parsing.');
+            }
+
+            if (isset($urlParts['path'])) {
+                $pathSegments = explode('/', $urlParts['path']);
+                $lastSegment = array_pop($pathSegments);
+                $lastSegment = rawurlencode($lastSegment);
+                $pathSegments[] = $lastSegment;
+
+                $urlParts['path'] = implode('/', $pathSegments);
+            }
+
+            $expected = $this->buildUrl($urlParts);
```

Comment:
What's the reason you change (encode) the `$expected` value like this for 6.8? It should be enough to change the expected values in the dataProvider `generateProvider` to be properly url encoded